### ISSUE TITLE
Django setting EMAIL_SUBJECT_PREFIX unused in communication emails

### ISSUE THREAD

**Issue by krzysztofjeziorny** [django-oscar/django-oscar#3708]
### Issue Summary

Setting `EMAIL_SUBJECT_PREFIX` ([docs](https://docs.djangoproject.com/en/3.2/ref/settings/#email-subject-prefix)) has no effect in emails send by Oscar.

### Steps to Reproduce

The email subjects are defined in `src/oscar/templates/oscar/communication/emails/*_subject.txt` templates. 

https://github.com/django-oscar/django-oscar/blob/1622f03fec1ba50905b750000c57999e909fc7a3/src/oscar/templates/oscar/communication/emails/commtype_order_placed_subject.txt#L1

I've tried creating my own template tag for the subject prefix, and also using Oscar's `{{ shop_name }}`

https://github.com/django-oscar/django-oscar/blob/1622f03fec1ba50905b750000c57999e909fc7a3/src/oscar/core/context_processors.py#L23

`{% load i18n %}{{ shop_name }}{% blocktrans with number=order.number %}Confirmation of order {{ number }}{% endblocktrans %`

but it it wasn't displayed either. Hard-coding the Prefix works, but is not so flexible. 

Am I missing something?

### Technical details

* Python version: 3.9.2
* Django version: 3.1.11
* Oscar version: 3.0

**Comment by solarissmoke** [comment]
No, you're not missing something - Oscar basically ignores that Django setting entirely. 

The context processor option won't work either, because context processors are not run when rendering email templates - only view templates. What you can do however is override the [`get_base_context()`](https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/communication/utils.py#L159-L163) method on the `Dispatcher`, to include context variables that you want to be available in all templates - i.e., insert `site_name` there and it will work in the way you want it to.

There's a separate question about whether Oscar should respect `EMAIL_SUBJECT_PREFIX`. I think adding that now would break backward-compatibility for projects that rely on it *not* being used. I personally think appending prefixes to all customer emails feels a bit weird/unnecessary for transactional emails from an e-commerce website (i.e., the sender email should be sufficient). This is just opinion though - open to input from others.
